### ISSUE TITLE
[a11y]: DataTable [with-radio-selection] - keyboard nav radio button bug prev/next

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14270]
### Package

@carbon/react

### Browser

Safari & iPhone

### Package version

@carbon/react@1.32.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO

### Description
First Column in the Table

Unable to move to the previous / next rows from the first column using the Row navigation Gesture in VO

Unable to move to the previous / next rows from the first column using the Row navigation Gesture in VO. 
Note: We are able to navigate through the Rows from other columns in the table.

### Recommended Fix
User should be able to navigate through the table Rows using the Row navigation gestures. 

When the standard keyboard navigation method is not supported, it impacts the user’s ability to interact with page controls. This affects screen reader users. 

### WCAG 2.1 Violation
2.1.1 Keyboard navigation

### Reproduction/example
Record `Number 76` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

https://react.carbondesignsystem.com/?path=/story/components-datatable-selection--with-radio-selection

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
